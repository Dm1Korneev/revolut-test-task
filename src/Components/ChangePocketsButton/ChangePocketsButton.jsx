import React from 'react';
import PropTypes from 'prop-types';

import { Button } from './ChangePocketsButton.styled';

const propTypes = {
  changePockets: PropTypes.func.isRequired,
};

function ChangePocketsButton(props) {
  const {
    changePockets,
  } = props;

  return (
    <Button type="button" onClick={changePockets} aria-label="Change pockets" />
  );
}

ChangePocketsButton.propTypes = propTypes;

export default ChangePocketsButton;
