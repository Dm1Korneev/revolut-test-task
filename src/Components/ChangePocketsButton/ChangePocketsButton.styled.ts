
import styled from 'styled-components';

type Props = {
    onClick: () => void;
  }

export const Button = styled.button<Props>`
    display: block;
    cursor: pointer;
    width: 100%;
    height: 100%;
    background-color: rgb(255, 255, 255);
    background: url('icons/import_export-24px.svg');
    background-repeat: no-repeat;
    background-position: 50% 50%;
    border-radius: inherit;
    transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
                box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
                border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;

    &:hover {
        background-color: rgb(250, 250, 250);
    }
`;
