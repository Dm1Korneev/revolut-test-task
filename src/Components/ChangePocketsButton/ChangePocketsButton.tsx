import React, { FC } from 'react';

import { Button } from './ChangePocketsButton.styled';

type Props = {
  changePockets: () => void;
}
const ChangePocketsButton: FC<Props> = (props): JSX.Element => {
  const {
    changePockets,
  } = props;

  return (
    <Button type="button" onClick={changePockets} aria-label="Change pockets" />
  );
};

export default ChangePocketsButton;
