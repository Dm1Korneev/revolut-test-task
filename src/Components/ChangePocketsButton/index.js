import commonHoc from 'Components/commonHoc';
import { changePockets } from 'Redux/actions';

import ChangePocketsButton from './ChangePocketsButton';

const mapDispatchToProps = {
  changePockets,
};

export default commonHoc(ChangePocketsButton, {
  mapDispatchToProps,
});
