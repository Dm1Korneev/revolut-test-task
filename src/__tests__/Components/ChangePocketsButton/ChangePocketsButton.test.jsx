import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import ChangePocketsButton from 'Components/ChangePocketsButton/ChangePocketsButton';

describe('ChangePocketsButton component', () => {
  const changePockets = jest.fn();
  const props = {
    changePockets,
  };

  test('renders the component', () => {
    const { asFragment } = render(<ChangePocketsButton {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('change value', () => {
    const { getByRole } = render(<ChangePocketsButton {...props} />);
    fireEvent.click(getByRole('button'));
    expect(changePockets).toHaveBeenCalledTimes(1);
  });
});
