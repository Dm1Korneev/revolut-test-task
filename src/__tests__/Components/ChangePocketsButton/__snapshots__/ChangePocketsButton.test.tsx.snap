// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePocketsButton component renders the component 1`] = `
<DocumentFragment>
  <button
    aria-label="Change pockets"
    class="sc-bdVaJa bXZQJU"
    type="button"
  />
</DocumentFragment>
`;
